src/main/java/ch/njol/util/coll/CollectionUtils.java

```diff
@@ -435,5 +437,26 @@ public static Double[] wrap(double[] primitive) {
 		}
 		return wrapped;
 	}
+
+	/**
+	 * Removes the first n elements from a Collection that matches the filter
+	 *
+	 * @param coll the {@link Collection} to remove from.
+	 * @param n the amount of elements to remove
+	 * @param filter the {@link Predicate} / filter that will be checked to remove the elements
+	 * @return amount of elements that matched the filter and have been removed.
+	 */
+	public <E> int removeFirstIf(Collection<E> coll, int n, Predicate<E> filter) {
```

Comment:
This method is not used with the current version of this PR. If you think it can still be useful in the future, it's fine to keep it, but otherwise change the first arg from a Collection to a List, as a collection isn't ordered, and therefore the `first n elements` doesn't make sense for all collections.

![image](https://user-images.githubusercontent.com/29547183/176716392-fb8c9c6c-334b-476c-aa6e-5295a28e1135.png)